### PR TITLE
Add options to enable Develocity

### PR DESCRIPTION
**Pull request by ppkarwasz**
This PR moves the changes from the experimental `develocity` branch to the `main` branch.

It features two main changes:

1. A new `reproducibility-check-enabled` option for the reusable builds. This option allow to disable reproducibility checks in some workflows.
2. A new `develocity-enabled` option for the reusable builds. This option allows to publish the result of the build to the https://ge.apache.org server.

### CODE REVIEW

File: .github/workflows/build-reusable.yaml [hunk 1]
```diff
@@ -67,6 +108,13 @@ jobs:
             -DinstallAtEnd=true \
             clean install
 
+      - name: Clean up Develocity
+        if: always()
```
Comment (vy) [inline .github/workflows/build-reusable.yaml:112]: Why don't we simply remove the `if` condition instead?

File: .github/workflows/build-reusable.yaml [hunk 2]
```diff
@@ -67,6 +108,13 @@ jobs:
...
+        shell: bash
+        run: |
+          rm -f .mvn/extensions.xml
+          echo "MAVEN_OPTS=" >> "$GITHUB_ENV"
```
Comment (vy) [inline .github/workflows/build-reusable.yaml:116]: Why do we truncate the `MAVEN_OPTS`?

File: .github/workflows/build-reusable.yaml [hunk 3]
```diff
@@ -100,19 +148,10 @@ jobs:
             --show-version --batch-mode --errors --no-transfer-progress \
             site
 
-      # We upload tests results if the build fails.
```
Comment (vy) [inline .github/workflows/build-reusable.yaml:103]: Why is this removed?

File: .github/workflows/build-reusable.yaml
Comment (vy) [inline .github/workflows/build-reusable.yaml:1]: Could you update `features.adoc#reusable-workflows` regarding the new Develocity changes, please?

File: .github/workflows/build-reusable.yaml (hunk 1 above)
Comment (ppkarwasz) [reply .github/workflows/build-reusable.yaml:112]: You are right, if the build fails, no need to clean it up.

Comment (ppkarwasz) [reply .github/workflows/build-reusable.yaml]: Fixed in https://github.com/apache/logging-parent/pull/229/commits/8975e12ce43007e9816e9996363e63e87c3ee477

File: .github/workflows/build-reusable.yaml (hunk 2 above)
Comment (ppkarwasz) [reply .github/workflows/build-reusable.yaml:117]: The `gradle/develocity-actions/maven-setup` action modifies the `MAVEN_OPTS` environment variable. We restore it to the default (_empty_) value, to prevent site generation from failing.

I added some comments and checks in https://github.com/apache/logging-parent/pull/229/commits/8975e12ce43007e9816e9996363e63e87c3ee477 and https://github.com/apache/logging-parent/pull/229/commits/5980ad9cc94f85718edef8f720eea50b49575696

File: .github/workflows/build-reusable.yaml (hunk 3 above)
Comment (ppkarwasz) [reply .github/workflows/build-reusable.yaml:103]: The test results are no longer necessary, since Develocity provides the same data in a more browseable format.

I restored this step partially in https://github.com/apache/logging-parent/pull/229/commits/bbd733b135586889bee24b62a4510f52e752c7f7, since Develocity does not upload the test log files generated under `target/logs`.

File: .github/workflows/build-reusable.yaml
Comment (ppkarwasz) [reply .github/workflows/build-reusable.yaml:1]: Fixed in https://github.com/apache/logging-parent/pull/229/commits/8975e12ce43007e9816e9996363e63e87c3ee477.

**Review by vy — APPROVED** [review]

**Comment by ppkarwasz** [comment]
After some minor changes, the workflow was used in a successful run: https://github.com/apache/logging-log4j2/actions/runs/10565985064
